### PR TITLE
Implemented AUAV absolute/differential pressure sensor support

### PR DESCRIPTION
**Pull request by alexcekay**
### Solved Problem
We want to be able to use the Amphenol AUAV pressure sensors. Those support measuring both absolute as well as differential pressure.

### Solution
- Added support for reading the absolute pressure measurements
- Added support for reading the differential pressure measurements
- Added a parameter to chose the exact type of AUAV sensor

### Changelog Entry
For release notes:
```
Feature Added Amphenol AUAV absolute/differential pressure sensor support
New parameter: SENS_EN_AUAVX
```

### Alternatives
When we want to use that sensor there is no alternative.

### Test coverage
- Was tested at the bench
- Will be tested on a VTOR with another pressure sensor attached for comparison

### CODE REVIEW

**Comment by sfuhrer** [comment]
Testflown on a small VTOL. 

AUAV 05: 
https://review.px4.io/plot_app?log=c4a8fc06-bb07-4234-a4d6-3900867ec92c
image

AUAV 10:
https://review.px4.io/plot_app?log=8324e5cb-de29-4eae-a237-897f8d65792e
image

The test setup was with one pitot tube connecting to both an AUAV and an SDP3X sensor through Y-tubes. I wouldn't rule out that this setup negatively impacts the data quality. In the logs the instances 0 of `differential_pressure` and `sensor_baro` are coming from the AUAV sensor, `differential_pressure.01` from the SDP3X and` sensor_baro.01/02` from the Skynode-internal baros.

Findings:
- AUAV05: differential pressure very close to readings of SDP3X in average, though much more noise, and even more noise on baro
- AUAV10: differential pressure has low noise level but has a relatively large scale offset to the SDP3X data. The baro pressure readings still spike but less than for the 05 sensor.

Temperature data from both sensors look clean - why is there though a difference between the one published by the AUAV on the baro from the one published on the differential_pressure topic? Do they not come from the same source?
image

**Review by niklaut — DISMISSED** [review]
Yes, driver looks clean, nice!

File: ROMFS/px4fmu_common/init.d/rc.sensors [hunk 1]
```diff
@@ -184,6 +184,18 @@ then
...
 
+# AUAV differential pressure sensor external I2C
+if param greater -s SENS_EN_AUAVX 0
+then
+	auav start -D -X
+fi
+
+# AUAV absolute pressure sensor external I2C
+if param greater -s SENS_EN_AUAVX 0
+then
```
[the remark this answers is not in the document]
Comment (alexcekay) [reply ROMFS/px4fmu_common/init.d/rc.sensors]: Yes of course, fixed it.

**Review by niklaut — APPROVED** [review]
Excellent!

**Pull request merged**